src/Controllers/UserController.cs

```diff
@@ -9,9 +9,8 @@ namespace magnus_backend.Controllers;
 
 [Route("api/user")]
 [ApiController]
-public class UserController(IUser user, IMongoClient mongoClient) : ControllerBase
+public class UserController(IMongoClient mongoClient) : ControllerBase, IUser
 {
```

Comment:
whats the difference between the before and after here?

Reply:
I was originally injecting a user into the controller, but that didnt make sense considering we are performing operations on user objects within this controller anyway. 

It originally came from me thinking that we needed a User `Service`, but we don't so I got rid of it as it was never used.

Reply:
Oh and its also inherits from two things now instead of one. `ControllerBase` and `IUser`. ControllerBase gives us general controller based methods and tools, and IUser is the User interfaces that tells us what User based functions we have to implement.